integtests/chatbot-api/kendra_workspace_test.py

```diff
@@ -0,0 +1,198 @@
+import boto3
+import json
+import time
+import uuid
+import pytest
+from clients.appsync_client import AppSyncClient
+
+
+@pytest.fixture(scope="module", autouse=True)
+def run_before_and_after_tests(client: AppSyncClient):
+    for workspace in client.list_workspaces():
+        if (
+            workspace.get("name") == "INTEG_TEST_KENDRA"
+            and workspace.get("status") == "ready"
+        ):
+            client.delete_workspace(workspace.get("id"))
+
+
+def test_create(client: AppSyncClient, default_embed_model):
+    rag_engines = client.list_rag_engines()
+    engine = next(i for i in rag_engines if i.get("id") == "kendra")
+    pytest.skip_flag = False
+    if engine.get("enabled") == False:
+        pytest.skip_flag = True
+        pytest.skip("Kendra is not enabled.")
+
+    kendra_indexes = client.list_kendra_indexes()
+    kendra_index_id = kendra_indexes[0].get("id")
+    pytest.workspace = client.create_kendra_workspace(
+        input={
+            "kind": "kendra",
+            "name": "INTEG_TEST_KENDRA",
+            "kendraIndexId": kendra_index_id,
+            "useAllData": True,
+        }
+    )
+
+    ready = False
+    retries = 0
+    # Wait for step function execution to complete and create the index
+    while not ready and retries < 10:
+        time.sleep(1)
+        retries += 1
+        workspace = client.get_workspace(pytest.workspace.get("id"))
+        if workspace.get("status") == "ready":
+            ready = True
+            break
+    assert ready == True
+
+
+def test_add_file(client: AppSyncClient):
+    if pytest.skip_flag == True:
+        pytest.skip("Kendra is not enabled.")
+    result = client.add_file(
+        input={
+            "workspaceId": pytest.workspace.get("id"),
+            "fileName": "content.txt",
+        }
+    )
+
+    s3_client = boto3.client("s3")
+    fields = result.get("fields")
+
+    # "fields" string format:
+    # {key=544aeaa9-3f3d-4995-843d-e66a993bf5e8/context.txt, AWSAccessKeyId=XXX
+    pairs = [pair.strip() for pair in fields.replace("{", "").split(',')]
+    fields_dict = dict(pair.split('=', 1) for pair in pairs)
+    bucket_name = result.get("url").split("//")[1].split(".")[0]
+    s3_client.put_object(
```

Comment:
Here you are using the credentials from the s3 client (current credentials). Ideally it should use the presign URL/credentials provided by this API

You should be able to pass the fields as is to a post request (without the s3 client)
https://github.com/boto/boto3/issues/1857

Reply:
Updated to use presigned URL and credentials provided by the API.